executor/aggfuncs/builder.go

```diff
@@ -198,7 +201,10 @@ func buildAvg(aggFuncDesc *aggregation.AggFuncDesc, ordinal int) AggFunc {
 		case mysql.TypeNewDecimal:
 			return &avgPartial4Decimal{baseAvgDecimal{base}}
 		case mysql.TypeDouble:
-			return &avgPartial4Float64{baseAvgFloat64{base}}
+			if ctx.GetSessionVars().WindowingUseHighPrecision {
+				return &avgPartial4Float64HighPrecision{baseAvgFloat64{base}}
+			}
+			return &avgPartial4Float64{avgPartial4Float64HighPrecision{baseAvgFloat64{base}}}
```

Comment:
Window function will only be used as `aggregation.CompleteMode`, so that this will never be hit.

Reply:
I will remove them.